lib/src/jpegr.cpp

```diff
@@ -628,6 +693,67 @@ uhdr_error_info_t JpegR::generateGainMap(uhdr_raw_image_t* sdr_intent, uhdr_raw_
   generateMap();
   std::for_each(workers.begin(), workers.end(), [](std::thread& t) { t.join(); });
 
+  /*refine_min_max_content_boost(gainmap_data, gainmap_min, gainmap_max, map_width, map_height,
+                               map_width, mUseMultiChannelGainMap);*/
+
+  float min_content_boost_log2 = gainmap_min[0];
+  float max_content_boost_log2 = gainmap_max[0];
+  for (int index = 1; index < (mUseMultiChannelGainMap ? 3 : 1); index++) {
+    min_content_boost_log2 = (std::min)(gainmap_min[index], min_content_boost_log2);
+    max_content_boost_log2 = (std::max)(gainmap_max[index], max_content_boost_log2);
+  }
+
+  std::function<void()> encodeMap = [this, gainmap_data, map_width, dest, min_content_boost_log2,
+                                     max_content_boost_log2, &jobQueue]() -> void {
+    size_t rowStart, rowEnd;
+
+    while (jobQueue.dequeueJob(rowStart, rowEnd)) {
+      if (mUseMultiChannelGainMap) {
+        for (size_t j = rowStart; j < rowEnd; j++) {
+          size_t dst_pixel_idx = j * dest->stride[UHDR_PLANE_PACKED] * 3;
+          size_t src_pixel_idx = j * map_width * 3;
+          for (size_t i = 0; i < map_width * 3; i++) {
+            reinterpret_cast<uint8_t*>(dest->planes[UHDR_PLANE_PACKED])[dst_pixel_idx + i] =
+                affineMapGain(gainmap_data[src_pixel_idx + i], min_content_boost_log2,
+                              max_content_boost_log2);
+          }
+        }
+      } else {
+        for (size_t j = rowStart; j < rowEnd; j++) {
+          size_t dst_pixel_idx = j * dest->stride[UHDR_PLANE_Y];
+          size_t src_pixel_idx = j * map_width;
+          for (size_t i = 0; i < map_width; i++) {
+            reinterpret_cast<uint8_t*>(dest->planes[UHDR_PLANE_Y])[dst_pixel_idx + i] =
+                affineMapGain(gainmap_data[src_pixel_idx + i], min_content_boost_log2,
+                              max_content_boost_log2);
+          }
+        }
+      }
+    }
+  };
+  workers.clear();
+  jobQueue.reset();
+  rowStep = threads == 1 ? map_height : 1;
+  for (int th = 0; th < threads - 1; th++) {
+    workers.push_back(std::thread(encodeMap));
+  }
+  for (size_t rowStart = 0; rowStart < map_height;) {
+    size_t rowEnd = (std::min)(rowStart + rowStep, map_height);
+    jobQueue.enqueueJob(rowStart, rowEnd);
+    rowStart = rowEnd;
+  }
+  jobQueue.markQueueForEnd();
+  encodeMap();
+  std::for_each(workers.begin(), workers.end(), [](std::thread& t) { t.join(); });
+
+  gainmap_metadata->max_content_boost = exp2(max_content_boost_log2);
```

Comment:
I'm afraid this may break the decoding on Android.

Could you help to encode two jpeg/r images, with and without this change and try displaying them on Android devices (or give me the images and I'll do the verification)